src/platform/bouffalolab/BL602/DiagnosticDataProviderImpl.cpp

```diff
@@ -248,36 +249,24 @@ CHIP_ERROR DiagnosticDataProviderImpl::GetWiFiBssId(ByteSpan & BssId)
 
 CHIP_ERROR DiagnosticDataProviderImpl::GetWiFiSecurityType(app::Clusters::WiFiNetworkDiagnostics::SecurityTypeEnum & securityType)
 {
-    using app::Clusters::WiFiNetworkDiagnostics::SecurityTypeEnum;
-    securityType = SecurityTypeEnum::kUnspecified;
-    // int authmode;
-
-    // authmode     = mgmr_get_security_type();
-    // securityType = MapAuthModeToSecurityType(authmode);
+    securityType = (app::Clusters::WiFiNetworkDiagnostics::SecurityTypeEnum) wifi_mgmr_get_security_type();
     return CHIP_NO_ERROR;
 }
 
 CHIP_ERROR DiagnosticDataProviderImpl::GetWiFiVersion(app::Clusters::WiFiNetworkDiagnostics::WiFiVersionEnum & wifiVersion)
 {
-    // TODO: Keeping existing behavior, but this looks broken.
-    // https://github.com/project-chip/connectedhomeip/issues/25546
-    wifiVersion = app::Clusters::WiFiNetworkDiagnostics::WiFiVersionEnum::kA;
-    return CHIP_NO_ERROR;
+    return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
 }
 
 CHIP_ERROR DiagnosticDataProviderImpl::GetWiFiChannelNumber(uint16_t & channelNumber)
 {
-    channelNumber = 0;
-
-    // channelNumber = mgmr_get_current_channel_num();
-
+    channelNumber = wifiMgmr.channel;
     return CHIP_NO_ERROR;
 }
 
 CHIP_ERROR DiagnosticDataProviderImpl::GetWiFiRssi(int8_t & rssi)
 {
-    // rssi = mgmr_get_rssi();
-
+    rssi = wifiMgmr.wlan_sta.sta.rssi;
```

Comment:
Again, does this do the right thing if we're not connected to wifi at all right now?

Reply:
Fixed.